nix/modules/rust.nix

```diff
@@ -21,78 +21,61 @@
 
       options.rustHelper = lib.mkOption { type = lib.types.raw; };
 
-      config.rustHelper = {
-        defaultTrack = "stable";
-        defaultVersion = "1.77.2";
-
-        defaultExtensions = [
-          "rust-src"
-          "rust-analyzer"
-          "clippy"
-          "rustfmt"
-        ];
-
-        defaultTargets = [
-          "wasm32-unknown-unknown"
-        ];
-
-        defaultStdenv = pkgs:
-          if pkgs.stdenv.isLinux
-          then pkgs.stdenvAdapters.useMoldLinker pkgs.stdenv
-          else pkgs.stdenv;
-
-        mkRustPkgs =
-          { track ? config.rustHelper.defaultTrack
-          , version ? config.rustHelper.defaultVersion
-          , extensions ? config.rustHelper.defaultExtensions
-          , targets ? config.rustHelper.defaultTargets
-          }:
-          import inputs.nixpkgs {
+      config.rustHelper =
```

Comment:
Bad diff, a summary of the changes:
- Make `mkRustPkgs` into a static `rustPkgs` which is just nixpkgs + the rust-overlay, rather than being dynamically built and trying to cleverly inject a Rust version with a second overlay
- Update `mkRust` to use the new `rustPkgs` and select a Rust version which used to be in `mkRustPkgs`

This resolves an infinite recursion problem. I traced the recursion problem to the `rust-overlay` but couldn't find any examples that use two overlays to set up the Rust version. I don't see a reason to do that when we can take Nix code from the tested/working examples in the `rust-overlay` project

Reply:
Why `rustPkgs` and not just `pkgs`? Is there a place where we need to access `pkgs` without the Rust overlay?

Reply:
Again, good point. I was trying to keep the naming that was here before but it's not necessary. No, this `pkgs` is local to this file now so it can't be used elsewhere. I'll rename 👍🏻

Reply:
Oh no, I was wrong, there is a reason. We're accepting `pkgs` as an input and modifying it, so we can't have a naming conflict there. So there is/was a reason for this